libs/vm-modules/src/ml/model/model.cpp

```diff
@@ -395,29 +417,34 @@ void VMModel::AssertLayerTypeMatches(SupportedLayerType                layer,
   };
   if (std::find(valids.begin(), valids.end(), layer) == valids.end())
   {
-    std::string const message{"Invalid params specified for \"" + LAYER_NAMES_.at(layer) +
-                              "\" layer."};
-    vm_->RuntimeError(message);
-    throw std::runtime_error(message);
+    throw std::runtime_error("Invalid params specified for \"" + LAYER_NAMES_.at(layer) +
```

Comment:
I think this was better before. The reason is that if we try to run a smart contract that uses this model function and fails here, the vm->RuntimeError(message) should route that useful message to std::out for the developer in a sensible way - whereas merely throwing will not give this information to the developer.

Reply:
if you like we can discuss this next week and I can show you how to run a simple smart contract and try it out for yourself. That might make this clearer.

Reply:
Runtime error will be called in a caller function, the top level .AddLayer<> contains a try-catch block that now covers all possible usages of ParseName and AddLayerSpecificImpl, so any exception will be caught and turned into VM-runtime error. So we still have VM logs with this error if occurs.

Reply:
I have replaced RuntimeErrors with exceptions wherever an immediate stop of the machine does not happen, e.g. where the control flow does not return to VM handlers. And vice-versa, in AddLayer and other etch-object-member-functions I replaced c++ exceptions with RuntimeErrors - because after return from this method control flow goes to VM hadler, the machine can be safely stopped and error log appears. Mixing both things in one code block did not provide expected result for me.